src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.MethodParametersInfo.cs

```diff
@@ -0,0 +1,297 @@
+                    .ToImmutableArray();
+
+                // 5. Parameter without any argument.
+                // Case 1: only parameter has default value would be left here. The parameter will be replaced by the default value.
+                // Case 2: there is no arguments and the parameter is params array. An array needs to be declared in the caller. Similarly to what is done in step 4
+                // Example for case 2:
+                // Before:
+                // void Caller(bool x)
+                // {
+                //     Callee();
+                // }
+                // void Callee(params int[] a)
+                // {
+                //     DoSomething(a);
+                // }
+                // After:
+                // void Caller(bool x)
+                // {
+                //     int[] a = { };
+                //     DoSomething(a);
+                // }
+                // void Callee(params int[] a)
+                // {
+                //     DoSomething(a);
+                // }
+                var parametersWithoutArgument = calleeMethodSymbol.Parameters.RemoveRange(allParametersWithArgument);
+                if (parametersWithoutArgument.Length == 1 && parametersWithoutArgument[0].IsParams)
+                {
+                    parametersNeedGenerateDeclarations = parametersNeedGenerateDeclarations.Concat((
+                        parametersWithoutArgument[0], ImmutableArray<SyntaxNode>.Empty));
+                }
+
+                var parametersWithDefaultValue = parametersWithoutArgument
+                    .Where(parameterSymbol => parameterSymbol.HasExplicitDefaultValue)
+                    .ToImmutableArray();
+
+                return new MethodParametersInfo(
+                    parametersWithIdentifier.SelectAsArray(parameterAndArgument =>
+                        (parameterAndArgument.parameterSymbol, parameterAndArgument.arguments[0])),
+                    parametersWithVariableDeclarationArgument.SelectAsArray(parameterAndArgument =>
+                        (parameterAndArgument.parameterSymbol, parameterAndArgument.arguments[0])),
+                    parametersNeedGenerateDeclarations,
+                    parametersWithDefaultValue,
+                    parametersWithLiteralArgument.SelectAsArray(parameterAndArgument =>
+                        (parameterAndArgument.parameterSymbol, parameterAndArgument.arguments[0])));
+            }
+
+            private static bool ParameterNeedsGenerateDeclarationFilter(
+                AbstractInlineMethodRefactoringProvider<TInvocationSyntaxNode, TExpressionSyntax, TArgumentSyntax> inlineMethodRefactoringProvider,
+                ISyntaxFacts syntaxFacts,
+                SemanticModel semanticModel,
+                (IParameterSymbol parameterSymbol, ImmutableArray<SyntaxNode> arguments) parameterAndArguments,
+                CancellationToken cancellationToken)
+            {
```

Comment:
1. why is this a static method her, instead of an instance method on AbstractInlineMethodRefactoringProvider?
2. no need for syntaxfacts to be passed along.